docs/sources/reference/commandline/cli.md

```diff
@@ -767,6 +769,46 @@ you refer to it on the command line.
 > children) for security reasons, and to ensure repeatable builds on remote
 > Docker hosts. This is also the reason why `ADD ../file` will not work.
 
+Sometimes building an image will require access to certain environment variables
+that are specific to the build host like http-proxy; user credentials for
+pulling intermediate files etc. Most likely these variables can't be persisted
+in the built image for portatbility or security reasons i.e. it's not desirable
+to use the 'ENV' primitive of the Dockerfile.
+
+    $ docker build --build-env HTTP_PROXY=http://10.20.30.2:1234 .
+
+This will allow passing the built-time environment variables, that can be
+accessed like regular environment variables in the 'RUN' primitive of the
+Dockerfile, without persisting them in the intermediate or final images.
+
+> **Note:** environment variables defined using 'ENV' primitive of Dockerfile
+> will override the build-time environment variables.
+
+> **Note:** When desirable, variables passed from command-line can be persisted
+> in the final image by using ENV primitive and build-time environment variables
+> as shown below.
+
+    $ docker build --build-env CONT_IMG_VER=v2.0.1 - <<MARK
+    -> FROM ubuntu
+    -> ENV CONT_IMG_VER ${CONT_IMG_VER:-v1.0.0}
+    -> MARK
```

Comment:
```
In the above example, the `ENV` command will do two things.  First, it will set `CONT_IMG_VER` to `v1.0.0` unless a build-time environment variable was defined for it.  Second, it will persist that variable in the resulting image.
```